doc/developers/release_process.rst

```diff
@@ -0,0 +1,311 @@
+
+1. Update docs. Note that this is for the final release, not necessarily for
+   the RC releases. These changes should be made in main and cherry-picked
+   into the release branch, only before the final release.
+
+   - Edit the ``doc/whats_new/v0.99.rst`` file to add release title and list of
+     contributors.
+     You can retrieve the list of contributor names with:
+
+     ::
+
+       $ git shortlog -s 0.98.33.. | cut -f2- | sort --ignore-case | tr '\n' ';' | sed 's/;/, /g;s/, $//' | fold -s
+
+     - For major releases, link the release highlights example from the ``doc/whats_new/v0.99.rst`` file.
+
+   - Update the release date in ``whats_new.rst``
+
+   - Edit the ``doc/templates/index.html`` to change the 'News' entry of the
+     front page (with the release month as well).
+
+2. On the branch for releasing, update the version number in
+   ``sklearn/__init__.py``, the ``__version__``.
+
+   For major releases, please add a 0 at the end: `0.99.0` instead of `0.99`.
+
+   For the first release candidate, use the `rc1` suffix on the expected final
+   release number: `0.99.0rc1`.
+
+3. Trigger the wheel builder with the ``[cd build]`` commit marker using
+   the command:
+
+   .. prompt:: bash $
+
+    git commit --allow-empty -m "Trigger wheel builder workflow: [cd build]"
+
+   The wheel building workflow is managed by GitHub Actions and the results be browsed at:
+   https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Wheel+builder%22
+
+.. note::
+
+  Before building the wheels, make sure that the ``pyproject.toml`` file is
+  up to date and using the oldest version of ``numpy`` for each Python version
+  to avoid `ABI <https://en.wikipedia.org/wiki/Application_binary_interface>`_
+  incompatibility issues. Moreover, a new line have to be included in the
+  ``pyproject.toml`` file for each new supported version of Python.
+
+.. note::
+
+  The acronym CD in `[cd build]` stands for `Continuous Delivery
+  <https://en.wikipedia.org/wiki/Continuous_delivery>`_ and refers to the
+  automation used to generate the release artifacts (binary and source
+  packages). This can be seen as an extension to CI which stands for
+  `Continuous Integration
+  <https://en.wikipedia.org/wiki/Continuous_integration>`_. The CD workflow on
+  GitHub Actions is also used to automatically create nightly builds and
+  publish packages for the development branch of scikit-learn. See
+  :ref:`install_nightly_builds`.
+
+4. Once all the CD jobs have completed successfully in the PR, merge it,
+   again with the `[cd build]` marker in the commit message. This time
+   the results will be uploaded to the staging area.
+
+   You should then be able to upload the generated artifacts (.tar.gz and .whl
+   files) to https://test.pypi.org using the "Run workflow" form for the
+   following GitHub Actions workflow:
+
+   https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Publish+to+Pypi%22
```

Comment:
Here we can maybe be more specific and warn to be careful to set the appropriate branch (main for a major/minor release and 0.999.X for a bugfix release)

Reply:
a2f71c6 should have clarified that.